graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/responses/IndexSetFieldType.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.rest.resources.system.indexer.responses;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.graylog2.rest.resources.entities.EntityAttribute;
+import org.graylog2.rest.resources.entities.EntityDefaults;
+import org.graylog2.rest.resources.entities.Sorting;
+
+import java.util.Comparator;
+import java.util.List;
+
+public record IndexSetFieldType(@JsonProperty(FIELD_NAME) String fieldName,
+                                @JsonProperty(TYPE) String type,
+                                @JsonProperty(IS_CUSTOM) boolean isCustom,
+                                @JsonProperty(IS_RESERVED) boolean isReserved) {
+
+    public static final String FIELD_NAME = "field_name";
```

Comment:
What about making these constants private?

Reply:
I will make them package-protected, because of the usage in tests. It is also possible that in the future they may need to go back to public, but we will see...